src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -57,12 +57,14 @@ public function handle($request, Closure $next)
 
         $this->storeCurrentUrl($request, $session);
 
-        $this->addCookieToResponse($response, $session);
+        if ($this->manager->driver()->isDirty()) {
+            $this->addCookieToResponse($response, $session);
```

Comment:
Shouldn't the cookie always be sent, irrespective whether the session is dirty or not?

Reply:
all and all, if we do not change the session, we also should not change the cookie as well.